plugins/outputs/nats/README.md

```diff
@@ -39,8 +39,35 @@ to use them.
   # credentials = "/etc/telegraf/nats.creds"
 
   ## NATS subject for producer messages
+  ## For jetstream this is also the subject where messages will be published
   subject = "telegraf"
 
+  ## Jetstream specific configuration. If specified, telegraf will use Jetstream to publish messages
+  # [outputs.nats.jetstream]
+    ## Specifies whether telegraf should create the stream at the startup or not. It will only create if it doesn't exist.
+    # auto_create_stream = true
+
+    ## Name of the stream where nats jetstream will publish the messages
+    # stream = "my-jetstream"
```

Comment:
auto_create_stream says it is true by default, but do you set this to true? What do you want the default to be - that is what should be in the docs. Should these even be an option and we can instead just autocreate for the user? Why would they not want to autocreate?

How about if a user set's the stream option, rename to `jetstream_stream`, then you enable jetsteam?

Then you can make the jetstream config its own table:

```toml
[[[inputs.nats]]
jetstream_stream = "mystream"
[inputs.nats.jetstream]
  retention = "workqueue"
  ...
```

Thoughts?

Reply:
There are cases when the stream already exists and users want to use that existing one.
But we can remove autocreate option, NATS provides us with a CreateOrUpdateStream API, which would fit this use case.

For the table, my previous approach is identical to what you suggested. However there are limitations to this, we will have to redefine the config structure in telegraf. There are about 26 options currently! And maintaining the struct in case of changes will be difficult.
Using existing structs from nats package (`jetstream.StreamConfig`) does not work, since there are custom types such as `RetentionPolicy` for which TOML unmarshalling doesn't work. 

JSON way seems flexible and NATS cli has that option using the config flag.

Reply:
> For the table, my previous approach is identical to what you suggested

Are you referring to this conversation: https://github.com/influxdata/telegraf/pull/14236#discussion_r1379281076

I do not think my comment was that you needed to expose all options. Rather, you should consider what may need to be exposed or what would be helpful. This is no different than on any other output, where the client can have many tunable parameters.

That said, I would not accept a JSON config inside a TOML-based tool ;) that is a nightmare for our users in terms of getting syntax correct.